tiledb/sm/c_api/tiledb_experimental.h

```diff
@@ -359,6 +359,36 @@ TILEDB_EXPORT int32_t tiledb_ctx_alloc_with_error(
     tiledb_ctx_t** ctx,
     tiledb_error_t** error) noexcept;
 
+/* ********************************* */
+/*           CONSOLIDATION           */
+/* ********************************* */
+
+/**
+ * Consolidates the given fragment URIs into a single fragment.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * const char* uris[2]={"__0_0_0807b1428b6c4ff48b3cdb3283ca7903_10",
+ *                      "__1_1_d9d965753d224194965575c1e9cdeeda_10"};
+ * tiledb_array_consolidate(ctx, "my_array", uris, 2);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param array_uri The name of the TileDB array whose metadata will
+ *     be consolidated.
+ * @param fragment_uris URIs of the fragments to consolidate.
+ * @param num_fragments Number of URIs to consolidate.
+ *
+ * @return `TILEDB_OK` on success, and `TILEDB_ERR` on error.
+ */
+TILEDB_EXPORT int32_t tiledb_array_consolidate_fragments(
```

Comment:
We need to document the caveats from the PR description here... This function sounds useful! (and it is, but people need to be aware even though it's in an optional include)

Reply:
Added some more documentation. I'm not sure if we should guard this API with a config option. I don't believe we have any other APIs that are doing this? @Shelnutt2 any thoughts?

Reply:
I'd vote to not guard it. This is for advanced use cases, mostly VCF being the immediate driver. The considerations will be reduced/go away once the next set of consolidation work is in place.

Reply:
Ok, sounds good to me. Resolving.